extensions/transform/v1/actions/parseNumberToTextWithDictionary/config/fields.ts

```diff
@@ -0,0 +1,42 @@
+import { z, type ZodTypeAny } from 'zod'
+import { type Field, FieldType } from '@awell-health/extensions-core'
+import { isEmpty, isNil } from 'lodash'
+
+export const fields = {
+  number: {
+    id: 'number',
+    label: 'Number',
+    description: 'The number you want to parse to a text (string)',
+    type: FieldType.NUMERIC,
+    required: true,
+  },
+  dictionary: {
+    id: 'dictionary',
+    label: 'Dictionary',
+    description:
+      'A JSON dictionary where each string key represents a number and maps to a corresponding string/text value',
+    type: FieldType.JSON,
+    required: true,
+  },
+} satisfies Record<string, Field>
+
+export const DictionarySchema = z
+  .string()
+  .transform((dict, ctx): Record<string, any> => {
+    if (isNil(dict) || isEmpty(dict)) return {}
+    try {
+      return JSON.parse(dict)
+    } catch (e) {
+      ctx.addIssue({
+        code: 'custom',
+        message:
+          'Invalid dictionary. Must be a valid key-value pair JSON object.',
+      })
+      return z.NEVER
+    }
+  })
+
+export const FieldsValidationSchema = z.object({
+  number: z.union([z.number(), z.nan()]),
```

Comment:
why not just z.number()

Reply:
Copy/pasted from another action without giving it much thought. But you're right, that should work too. Doing the change.

Reply:
Tested it, and apparently, it doesn't work. `z.number()` doesn't allow `NaN` value.

<img width="831" alt="Screenshot 2024-01-19 at 10 00 01" src="https://github.com/awell-health/awell-extensions/assets/5594064/be0a49f4-1eaf-48ff-9b87-a1210c0a8b99">